### PR TITLE
TETP-17 Employer can delete posting and base dialog component

### PR DESCRIPTION
**Pull request by pekkvil**
…ete posting mutation

### CODE REVIEW

**Review by mattikl — CHANGES_REQUESTED** [review]
The shared `Modal` component is also used by https://github.com/City-of-Helsinki/yjdh/blob/develop/frontend/benefit/handler/src/components/applicationReview/actions/handlingApplicationActions/HandlingApplicationActions.tsx#L70 so prop `cancelButtonLabel` needs to be added.

Otherwise looks good, so good to merge after adding this, fixing the merge conflict and seeing that all tests pass.

File: frontend/shared/src/components/modal/Modal.tsx
Comment (nenonja) [inline frontend/shared/src/components/modal/Modal.tsx:70]: It's not absolutely necessary to parameterize every text in shared components, if just use different texts in different application. So if you use shared Modal component, just bring translation with using the same key in your service.

Pros: less properties for shared component
Cons: have to use exactly the same key.

File: frontend/shared/src/components/toast/show-success-toast.ts
Comment (nenonja) [inline frontend/shared/src/components/toast/show-success-toast.ts:1]: Another option would be to rename current `show-error-toast` to `error-toast` and add `type` as required parameter, so we could use same component for `type: 'error'` and `type: 'succes'`. Of course it needs some refactoring which can be daunting job...

File: frontend/tet/admin/src/pages/_app.tsx
Comment (nenonja) [inline frontend/tet/admin/src/pages/_app.tsx:24]: `<ConfirmDialog />` ?

File: frontend/tet/admin/src/pages/_document.tsx
Comment (nenonja) [inline frontend/tet/admin/src/pages/_document.tsx:1]: There is `ServerDocment` in shared code which perhaps could be used?... just the portal part should rendered only optionally, if some new property (ie `portal:boolean`) is given.

File: frontend/tet/admin/src/store/DialogContext.tsx
Comment (nenonja) [inline frontend/tet/admin/src/store/DialogContext.tsx:46]: `const content = action?.payload?.content ?? ''`

**Review by nenonja — COMMENTED** [review]
Nice job with the modal! I probably also need confirmation dialog soon in kesaseteli handler-ui so it's nice addition. Some small somments added

**Review by mattikl — APPROVED** [review]

File: frontend/tet/admin/src/pages/_document.tsx
Comment (pekkvil) [reply frontend/tet/admin/src/pages/_document.tsx:1]: Ok, thanks! I missed the shared option. I'll check if that could be used.

File: frontend/shared/src/components/toast/show-success-toast.ts
Comment (pekkvil) [reply frontend/shared/src/components/toast/show-success-toast.ts:1]: I also thought about using multi purpose component but chose the easy way instead and used the same convention that was used of the show-success

**Comment by pekkvil** [comment]
Failing back end checks in review enviroment but no back end changes in pr

**Pull request merged**
